index.js

```diff
@@ -1144,23 +1144,11 @@ exports.main = function() {
     // set's up the addon for dev mode.
     overrideDefaults();
 
-    // if the user has seen at least step 1, we need to add the ActionButton
-    // now, or else the code in the following conditional could try to show
-    // a notification to the user but, this will error because allAboard is undefined.
-    // Also ensure that the user hasn't hit the last sidebar, because we can't get that info through sidebarProps
-    if (typeof simpleStorage.step !== 'undefined' && simpleStorage.step < 5) {
-        addAddOnButton();
-        sidebarProps = getSidebarProps();
-        modifyAboutHome(sidebarProps.track, sidebarProps.step);
-    } 
-    // edge case time: If simpleStorage.step is undefined, it means the user has not seen
-    // even our first sidebar. This also means that simpleStorage.lastSidebarLaunchTime will
-    // be undefined so, no need to check that. The user might however have answered the
-    // initial on-boarding questions and then closed Firefox(or it crashed :-/). This means that
-    // if simpleStorage.step is undefined but, simpleStorage.isOnBoarding is not, start the
-    // notification timer, and add the add-on button to the chrome.
-    // Also ensure that the user hasn't hit the last sidebar, because we can't get that info through sidebarProps
-    else if (typeof simpleStorage.isOnBoarding !== 'undefined' && simpleStorage.step < 5) {
+    // if the user has seen at least step 1, and we aren't already to the reward sidebar
+    // OR if the user is onboarding and hasn't seen a step yet or they aren't to the 5th step,  
+    // add the get the sidebar props for the current step, add the actionButton, and modify about:home
+    if ((typeof simpleStorage.step !== 'undefined' && simpleStorage.step < 5) || (typeof simpleStorage.isOnBoarding !== 'undefined' && 
+        (simpleStorage.step < 5 || typeof simpleStorage.step === 'undefined'))) {
```

Comment:
Two things with this:
1. Needed to add logic in the else statement to account for simpleStorage.step being undefined. Without this, users will never get the badge again if they reopen firefox without engaging a CTA
2. I realized both the if and else do the same thing here, so I just added the else as an OR condition